adapter.go

```diff
@@ -335,11 +394,13 @@ const (
 )
 
 func NewAdapter(reader KeyReaderAt, opts ...AdapterOption) (*Adapter, error) {
+	_, okStream := reader.(KeyStreamerAt)
 	bc := &Adapter{
 		blockSize:       DefaultBlockSize,
 		numCachedBlocks: DefaultNumCachedBlocks,
 		reader:          reader,
 		splitRanges:     false,
+		canStream:       okStream,
```

Comment:
Instead of `canStream`, I'd rename `reader` to `keyReader` and add a `keyStreamer` member. In the rest of the code you'd check for non-nilness of keyStreamer and use it directly  without type-casting